src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Services/AzureSqlManagedInstanceAdapter.cs

```diff
@@ -0,0 +1,207 @@
+        {
+            Communicator.Remove(resourceGroupName, managedInstanceName);
+        }
+
+        /// <summary>
+        /// Convert a Management.Sql.LegacySdk.Models.ManagedInstance to AzureSqlDatabaseManagedInstanceModel
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group the managed instance is in</param>
+        /// <param name="resp">The management client managed instance response to convert</param>
+        /// <returns>The converted managed instance model</returns>
+        private static AzureSqlManagedInstanceModel CreateManagedInstanceModelFromResponse(Management.Sql.Models.ManagedInstance resp)
+        {
+            AzureSqlManagedInstanceModel managedInstance = new AzureSqlManagedInstanceModel();
+
+            // Extract the resource group name from the ID.
+            // ID is in the form:
+            // /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rgName/providers/Microsoft.Sql/managedInstances/managedInstanceName
+            string[] segments = resp.Id.Split('/');
+            managedInstance.ResourceGroupName = segments[4];
+
+            managedInstance.ManagedInstanceName = resp.Name;
+            managedInstance.Id = resp.Id;
+            managedInstance.FullyQualifiedDomainName = resp.FullyQualifiedDomainName;
+            managedInstance.AdministratorLogin = resp.AdministratorLogin;
+            managedInstance.Location = resp.Location;
+            managedInstance.Tags = TagsConversionHelper.CreateTagDictionary(TagsConversionHelper.CreateTagHashtable(resp.Tags), false);
+            managedInstance.Identity = resp.Identity;
+            managedInstance.FullyQualifiedDomainName = resp.FullyQualifiedDomainName;
+            managedInstance.SubnetId = resp.SubnetId;
+            managedInstance.LicenseType = resp.LicenseType;
+            managedInstance.VCores = resp.VCores;
+            managedInstance.StorageSizeInGB = resp.StorageSizeInGB;
+
+            Management.Internal.Resources.Models.Sku sku = new Management.Internal.Resources.Models.Sku();
+            sku.Name = resp.Sku.Name;
+            sku.Tier = resp.Sku.Tier;
+
+            managedInstance.Sku = sku;
+
+            return managedInstance;
+        }
+
+        /// <summary>
+        /// Convert a <see cref="System.Security.SecureString"/> to a plain-text string representation.
+        /// This should only be used in a proetected context, and must be done in the same logon and process context
+        /// in which the <see cref="System.Security.SecureString"/> was constructed.
+        /// </summary>
+        /// <param name="secureString">The encrypted <see cref="System.Security.SecureString"/>.</param>
+        /// <returns>The plain-text string representation.</returns>
+        [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
+        internal static string Decrypt(SecureString secureString)
```

Comment:
Use this instead: https://github.com/Azure/azure-powershell/blob/preview/src/Common/Commands.Common/Utilities/ConversionUtilities.cs#L95